### ISSUE TITLE
Support responsively skipping tick labels

### ISSUE THREAD

**Issue by jackwilsdon** [plouc/nivo#260]
A lot of libraries support skipping ticks at smaller screen sizes so that their labels do not overlap, however it doesn't seem that `nivo` does (taking the example `ResponsiveLine` down to a really small size keeps all X-axis labels).

It'd be nice to see this as it would make the responsive graphs a lot more mobile friendly.

**Comment by plouc** [comment]
@jackwilsdon, sorry I'm not planing to add this feature to nivo as it's quite difficult to guess what's crucial for the chart and which datum should be skipped.
If you want to achieve that kind of functionality, you can probably configure tick values according to size, using [that kind of library](https://github.com/contra/react-responsive).

**Comment by sebastienbarre** [comment]
@plouc But you do provide an option to *not* render the label inside the bars of a bar chart for example, using `labelSkipWidth`, so you must know the width of each bar at that point. The labels on the x-axis occupy the same space horizontally (except they are placed below the bars), so the same type of option could allow people to at least skip rendering of the axis labels if the available width is below a threshold. 

How about updating the `format` callback on the axis, so that it sends more than just the value to format for a given tick. Say, `format(value, { width, index, ... })` so that the callback receives the available width for that label, and maybe the index of that bar. With that piece of information, I could decide to render every-other label if the width is below a specific threshold (by returning an empty string as my formatted value for that tick).

In other words, let us take care of the work of deciding if the datum should be skipped by giving us the information that you know (but that we don't), such as the available width for that tick, and what the id or index of the datum at that tick.

Thanks for your work.
